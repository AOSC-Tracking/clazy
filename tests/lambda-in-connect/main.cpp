#include <QtCore/QObject>

struct A
{
    int v;
};


void test()
{
    QObject *o;
    int a, b, c;
    auto f = [&a]() {}; // OK
    o->connect(o, &QObject::destroyed, [a]() {}); // OK
    o->connect(o, &QObject::destroyed, [&a]() {}); // Warning
    QObject::connect(o, &QObject::destroyed, [&a]() { }); // Warning
    QObject::connect(o, &QObject::destroyed, [&]() { a; b; }); // Warning
    QObject::connect(o, &QObject::destroyed, [=]() { a; b; }); // OK

    A *a1;
    QObject::connect(o, &QObject::destroyed, [a1]() { a1->v; }); // OK
    QObject::connect(o, &QObject::destroyed, [&a1]() { a1->v; }); // Warning
}


static int s;

struct C
{

    void foo()
    {
        QObject *o;
        QObject::connect(o, &QObject::destroyed, [this]() { }); // OK
        QObject::connect(o, &QObject::destroyed, []() { s; }); // OK
    }

    int m;
};
