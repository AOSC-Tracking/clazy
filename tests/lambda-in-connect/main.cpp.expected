lambda-in-connect/main.cpp:15:42: warning: capturing local variable by reference in lambda [-Wclazy-lambda-in-connect]
lambda-in-connect/main.cpp:16:48: warning: capturing local variable by reference in lambda [-Wclazy-lambda-in-connect]
lambda-in-connect/main.cpp:17:54: warning: capturing local variable by reference in lambda [-Wclazy-lambda-in-connect]
lambda-in-connect/main.cpp:17:57: warning: capturing local variable by reference in lambda [-Wclazy-lambda-in-connect]
lambda-in-connect/main.cpp:22:48: warning: capturing local variable by reference in lambda [-Wclazy-lambda-in-connect]
lambda-in-connect/main.cpp:37:52: warning: capturing local variable by reference in lambda [-Wclazy-lambda-in-connect]
